knpu/Challenges/BddFeatures/RegistrationFeatureCoding.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Challenges\Interfaces;
+
+use KnpU\ActivityRunner\Activity\CodingChallenge\CodingContext;
+use KnpU\ActivityRunner\Activity\CodingChallenge\CorrectAnswer;
+use KnpU\ActivityRunner\Activity\CodingChallengeInterface;
+use KnpU\ActivityRunner\Activity\CodingChallenge\CodingExecutionResult;
+use KnpU\ActivityRunner\Activity\CodingChallenge\FileBuilder;
+use KnpU\ActivityRunner\Activity\Exception\GradingException;
+
+class RegistrationFeatureCoding implements CodingChallengeInterface
+{
+    /**
+     * @return string
+     */
+    public function getQuestion()
+    {
+        return <<<EOF
+You've assembled a crack team for a brand new project:
+a site where paleontologists and other dinosaur scientists
+can send each other important scientific messages, along
+with the usual cat photos.
+
+Each scientist will need to register to gain access, and
+you - knowing that using BDD will create a better product -
+are writing a registration feature. Fill in the 4 feature
+lines in `registration.feature`.
+EOF;
+    }
+
+    public function getFileBuilder()
+    {
+        $fileBuilder = new FileBuilder();
+        $fileBuilder
+            ->addFileContents('registration.feature', <<<EOF
+EOF
+            )
+            ->setEntryPointFilename('registration.feature')
+        ;
+
+        return $fileBuilder;
+    }
+
+    public function getExecutionMode()
+    {
+        return self::EXECUTION_MODE_PHP_NORMAL;
+    }
+
+    public function setupContext(CodingContext $context)
+    {
+    }
+
+    public function grade(CodingExecutionResult $result)
+    {
+        throw new GradingException('Oops...');
```

Comment:
@weaverryan I wondering how to grading features. There're any best solution for it?